lib/streamlit/elements/beta_vega_lite.py

```diff
@@ -0,0 +1,196 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""A Python wrapper around Vega-Lite."""
+
+import json
+from typing import cast
+
+import streamlit
+import streamlit.elements.arrow as arrow
+import streamlit.elements.lib.dicttools as dicttools
+from streamlit.logger import get_logger
+from streamlit.proto.BetaVegaLiteChart_pb2 import (
+    BetaVegaLiteChart as BetaVegaLiteChartProto,
+)
+
+LOGGER = get_logger(__name__)
+
+
+class BetaVegaLiteMixin:
+    def beta_vega_lite_chart(
+        self,
+        data=None,
+        spec=None,
+        use_container_width=False,
+        **kwargs,
+    ):
+        """Display a chart using the Vega-Lite library.
+
+        Parameters
+        ----------
+        data : pandas.DataFrame, pandas.Styler, numpy.ndarray, Iterable, dict, or None
+            Either the data to be plotted or a Vega-Lite spec containing the
+            data (which more closely follows the Vega-Lite API).
+
+        spec : dict or None
+            The Vega-Lite spec for the chart. If the spec was already passed in
+            the previous argument, this must be set to None. See
+            https://vega.github.io/vega-lite/docs/ for more info.
+
+        use_container_width : bool
+            If True, set the chart width to the column width. This takes
+            precedence over Vega-Lite's native `width` value.
+
+        **kwargs : any
+            Same as spec, but as keywords.
+
+        Example
+        -------
+
+        >>> import pandas as pd
+        >>> import numpy as np
+        >>>
+        >>> df = pd.DataFrame(
+        ...     np.random.randn(200, 3),
+        ...     columns=['a', 'b', 'c'])
+        >>>
+        >>> st.beta_vega_lite_chart(df, {
+        ...     'mark': {'type': 'circle', 'tooltip': True},
+        ...     'encoding': {
+        ...         'x': {'field': 'a', 'type': 'quantitative'},
+        ...         'y': {'field': 'b', 'type': 'quantitative'},
+        ...         'size': {'field': 'c', 'type': 'quantitative'},
+        ...         'color': {'field': 'c', 'type': 'quantitative'},
+        ...     },
+        ... })
+
+        Examples of Vega-Lite usage without Streamlit can be found at
+        https://vega.github.io/vega-lite/examples/. Most of those can be easily
+        translated to the syntax shown above.
+
+        """
+        proto = BetaVegaLiteChartProto()
+        marshall(
+            proto,
+            data,
+            spec,
+            use_container_width=use_container_width,
+            **kwargs,
+        )
+        return self.dg._enqueue("beta_vega_lite_chart", proto)
+
+    @property
+    def dg(self) -> "streamlit.delta_generator.DeltaGenerator":
+        """Get our DeltaGenerator."""
+        return cast("streamlit.delta_generator.DeltaGenerator", self)
+
+
+def marshall(proto, data=None, spec=None, use_container_width=False, **kwargs):
```

Comment:
Add types to parameters and return value

Reply:
Addressed